src/core/web-server.cc

```diff
@@ -530,6 +531,230 @@ class CDExecutor : public PCSX::WebExecutor {
+
+        if (request.method == PCSX::RequestData::Method::HTTP_HTTP_GET) {
+            if (path == "usage") {
+                nlohmann::json j;
+                for (uint32_t i = 0; i < 10; ++i) {
+                    j["slots"][i] = PCSX::g_gui->getSaveStateExists(i);
+                }
+                const auto& namedSaves = PCSX::g_gui->getNamedSaveStates();
+                for (uint32_t i = 0; i < namedSaves.size(); ++i) {
+                    const auto& filenamePair = namedSaves[i];
+                    j["named"][i]["name"] = filenamePair.second;
+                    j["named"][i]["filepath"] = filenamePair.first.string();
+                }
+                write200(client, j);
+                return true;
+            } else if (path == "load" || path == "save" || path == "delete") {
+                auto vars = parseQuery(request.urlData.query);
+                auto islot = vars.find("slot");
+                auto iname = vars.find("name");
+                if ((islot == vars.end()) && (iname == vars.end())) {
+                    client->write("HTTP/1.1 400 Bad Request\r\n\r\n");
+                    return true;
+                }
+                if ((islot != vars.end()) && (iname != vars.end())) {
+                    client->write("HTTP/1.1 400 Bad Request\r\n\r\n");
+                    return true;
+                }
+                if (islot != vars.end()) {
+                    std::string message;
+                    int slot = -1;
+                    try {
+                        slot = std::stoul(islot->second);
+                    } catch (std::exception const& ex) {
+                        message = fmt::format(
+                            "HTTP/1.1 400 Bad Request\r\n\r\nFailed to parse state slot value \"{}\".", islot->second);
+                        client->write(std::move(message));
+                        return true;
+                    }
+                    if (slot < 0 || slot >= 10) {
+                        message =
+                            fmt::format("HTTP/1.1 400 Bad Request\r\n\r\nState slot index {} out of range 0-9.", slot);
+                    } else {
+                        bool success = false;
+                        if (path == "load") {
+                            success = PCSX::g_gui->loadSaveStateSlot(slot);
+                        } else if (path == "save") {
+                            success = PCSX::g_gui->saveSaveStateSlot(slot);
+                        } else if (path == "delete") {
+                            success = PCSX::g_gui->deleteSaveStateSlot(slot);
+                        }
+                        if (success) {
+                            message =
+                                fmt::format("HTTP/1.1 200 OK\r\n\r\nState slot index {} {} successful.", slot, path);
+                        } else {
+                            message = fmt::format("HTTP/1.1 400 Bad Request\r\n\r\nState slot index {} {} failed.",
```

Comment:
*Technically* that's a 500 Internal Server Error :) The request was fine, the server didn't manage to handle it properly.